lib/new-settings-loader.js

```diff
@@ -4,14 +4,34 @@ const migration = require('./config-migration')
 
 const OLD_SETTINGS_LOADER_SCHEMA_VERSION = 1
 const NEW_SETTINGS_LOADER_SCHEMA_VERSION = 2
+const PASSWORD_FILLED = 'PASSWORD_FILLED'
+const SECRET_FIELDS = [
+  'bitgo.BTCWalletPassphrase',
+  'bitgo.LTCWalletPassphrase',
+  'bitgo.ZECWalletPassphrase',
+  'bitgo.BCHWalletPassphrase',
+  'bitgo.DASHWalletPassphrase',
+  'bitstamp.secret',
+  'infura.apiSecret',
+  'itbit.clientSecret',
+  'kraken.privateKey',
+  'twilio.authToken'
+]
 
 const accountsSql = `update user_config set data = $2, valid = $3, schema_version = $4 where type = $1;
 insert into user_config (type, data, valid, schema_version)
 select $1, $2, $3, $4 where $1 not in (select type from user_config)`
-function saveAccounts (accountsToSave) {
+function saveAccounts (accounts) {
   return loadAccounts()
     .then(currentAccounts => {
-      const newAccounts = _.assign(currentAccounts, accountsToSave)
+      const serviceCodes = _.keys(accounts)
+      const filledAccountInfo = _.map(code => { return { [code]: _.omitBy(_.isEmpty, accounts[code]) } }, serviceCodes)
```

Comment:
I think we should only apply this if the code is in the SECRET_FIELDS list, or else ops would never be able to delete any other config.

Reply:
Also when you need to return a object in a lamda function you can do this:

```js
code => ({ [code]: ..........  })
```

Reply:
Probably easier to just map over the original array isn't it? I don't see the reason to get the keys and then map over transforming into a object.

Reply:
Essentially, I wanted to map through each nested field of each account and remove what's empty. There's probably a better way of doing it all at once by accessing the nested fields in a single map but it was not being obvious to me.